.github/actions/perform_tests/action.yml

```diff
@@ -0,0 +1,31 @@
+name: 'Perform Tests'
+description: 'This action contains all the usual steps needed to perform the unit tests'
+
+inputs:
+  needs-shared-tests:
+    description: 'Flag to determine whether the action needs to perform the Shared  module tests'
+    required: true
+  needs-storage-tests:
+    description: 'Flag to determine whether the action needs to perform the Storage module tests'
+    required: true
+  needs-client-tests:
+    description: 'Flag to determine whether the action needs to perform the Client module tests'
+    required: true
+  github-token:
+    description: 'The Github Token'
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Run Shared tests if needed
```

Comment:
Nice clean way of checking what to test using the bundles under testing 👏 

Minor thing: I think this description needs to be updated since this now executes all needed tests.

Reply:
💯 ! missed